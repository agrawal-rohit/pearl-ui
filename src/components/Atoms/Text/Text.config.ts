export default {
  baseStyle: {
    color: {
      light: "neutral.900",
      dark: "neutral.50",
    },
    scaleFontSize: true,
  },
  variants: {
    h1: {
      fontFamily: "heading",
      fontWeight: "bold",
      fontSize: "8xl",
      lineHeight: "10xl",
    },
    h2: {
      fontFamily: "heading",
      fontWeight: "bold",
      fontSize: "5xl",
      lineHeight: "7xl",
    },
    t1: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "2xl",
      lineHeight: "4xl",
    },
    t2: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "l",
      lineHeight: "2xl",
    },
    st1: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "m",
      lineHeight: "xl",
    },
    st2: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "s",
      lineHeight: "l",
    },
    p1: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "m",
      lineHeight: "xl",
    },
    p2: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "s",
      lineHeight: "l",
    },
    caption: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "xs",
      lineHeight: "m",
    },
    btn1: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "m",
      lineHeight: "xl",
    },
    btn2: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "s",
      lineHeight: "m",
    },
    btn3: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "xs",
      lineHeight: "s",
    },
    btn4: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "2xs",
      lineHeight: "xs",
    },
  },
  defaults: {
    variant: "p1",
  },
};
