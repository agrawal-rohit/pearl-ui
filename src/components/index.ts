// Atoms
export { default as Box, BoxProps } from "./atoms/box/box";
export { default as Center } from "./atoms/center/center";
export { default as Collapse, CollapseProps } from "./atoms/collapse/collapse";
export { default as Divider, DividerProps } from "./atoms/divider/divider";
export { default as Fade, FadeProps } from "./atoms/fade/fade";
export { default as Grid, GridProps } from "./atoms/grid/grid";
export { default as Icon, IconProps } from "./atoms/icon/icon";
export {
  default as Pressable,
  PressableProps,
} from "./atoms/pressable/pressable";
export {
  default as ScaleFade,
  ScaleFadeProps,
} from "./atoms/scale-fade/scale-fade";
export {
  default as List,
  OrderedList,
  UnorderedList,
  ListProps,
} from "./atoms/list/list";
export { default as Screen, ScreenProps } from "./atoms/screen/screen";
export { default as Skeleton, SkeletonProps } from "./atoms/skeleton/skeleton";
export {
  default as SkeletonCircle,
  SkeletonCircleProps,
} from "./atoms/skeleton/skeleton-circle";
export {
  default as SkeletonText,
  SkeletonTextProps,
} from "./atoms/skeleton/skeleton-text";
export { default as Slide, SlideProps } from "./atoms/slide/slide";
export {
  default as SlideFade,
  SlideFadeProps,
} from "./atoms/slide-fade/slide-fade";
export { default as Spacer } from "./atoms/spacer/spacer";
export { default as Spinner, SpinnerProps } from "./atoms/spinner/spinner";
export {
  default as Stack,
  HStack,
  VStack,
  ZStack,
  HStackProps,
  VStackProps,
  StackProps,
} from "./atoms/stack/stack";
export { default as Text, TextProps } from "./atoms/text/text";

// Molecules
export { default as Avatar, AvatarProps } from "./molecules/avatar/avatar";
export {
  default as AvatarGroup,
  AvatarGroupProps,
} from "./molecules/avatar/avatar-group";
export { default as Badge, BadgeProps } from "./molecules/badge/badge";
export {
  default as withBadge,
  WithBadgeOptions,
} from "./molecules/badge/withBadge";
export { default as Button, ButtonProps } from "./molecules/button/button";
export {
  default as ButtonGroup,
  ButtonGroupProps,
} from "./molecules/button/button-group";
export {
  default as CheckBox,
  CheckBoxProps,
} from "./molecules/checkbox/checkbox";
export {
  default as CheckBoxGroup,
  CheckBoxGroupProps,
} from "./molecules/checkbox/checkbox-group";
export {
  default as IconButton,
  IconButtonProps,
} from "./molecules/icon-button/icon-button";
export { default as Image, ImageProps } from "./molecules/image/image";
export { default as Input, InputProps } from "./molecules/input/input";
export {
  default as PinInput,
  PinInputProps,
} from "./molecules/pin-input/pin-input";
export {
  default as Progress,
  ProgressProps,
} from "./molecules/progress/progress";
export { default as Radio, RadioProps } from "./molecules/radio/radio";
export {
  default as RadioGroup,
  RadioGroupProps,
} from "./molecules/radio/radio-group";
export { default as Switch, SwitchProps } from "./molecules/switch/switch";
export {
  default as Textarea,
  TextareaProps,
} from "./molecules/textarea/textarea";
export { default as Video, VideoProps } from "./molecules/video/video";
export {
  default as Accordion,
  AccordionProps,
  AccordionItemProps,
  AccordionButtonProps,
  AccordionPanelProps,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./molecules/accordion/accordion";
export { default as Alert, AlertProps } from "./molecules/alert/alert";
