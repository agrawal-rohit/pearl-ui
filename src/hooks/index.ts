export { useColorModeValue } from "./useColorModeValue";
export { useComponentConfig } from "./useComponentConfig";
export { useStyledProps } from "./useStyledProps";
export { useTheme } from "./useTheme";
