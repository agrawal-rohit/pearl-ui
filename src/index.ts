export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
