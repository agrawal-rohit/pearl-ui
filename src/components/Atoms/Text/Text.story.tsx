import React from "react";
import { storiesOf } from "@storybook/react-native";
import Text from "./Text";
import Screen from "../Screen/Screen";
import Box from "../Box/Box";

storiesOf("Typography", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("All", () => (
    <Box flex={1}>
      <Text variant="h1">Heading 1</Text>
      <Text variant="h2">Heading 2</Text>
      <Text variant="h3">Heading 3</Text>
      <Text variant="h4">Heading 4</Text>
      <Text variant="h5">Heading 5</Text>
      <Text variant="h6">Heading 6</Text>
      <Text variant="p1">Paragraph 1</Text>
      <Text variant="p2">Paragraph 2</Text>
      <Text variant="p3">Paragraph 3</Text>
      <Text variant="p4">Paragraph 4</Text>
      <Text variant="label">Label</Text>
      <Text variant="caption">Caption</Text>
      <Box backgroundColor="primary.500">
        <Text variant="btn1">Button 1</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn2">Button 2</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn3">Button 3</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn4">Button 4</Text>
      </Box>
    </Box>
  ));
