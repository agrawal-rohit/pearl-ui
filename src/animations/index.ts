import { useAnimationState } from "../hooks/useAnimationState";

/**
 * Custom hook for creating a fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from an opacity of 0 to an opacity of 1.
 * This creates a fade-in effect.
 *
 * @returns An animation state that creates a fade-in effect.
 */
export const useFade = () => {
  // Define the initial and final states for the fade animation.
  // The animation will start with an opacity of 0 (completely transparent) and end with an opacity of 1 (completely opaque).
  const fadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
    },
    to: {
      opacity: 1,
    },
  });

  // Return the fade animation state.
  return fadeAnimationState;
};

/**
 * Custom hook for creating a scale and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scale of 0.5 and opacity of 0 to a scale of 1 and opacity of 1.
 * This creates a combined effect of scaling and fading in.
 *
 * @returns An animation state that creates a scale and fade-in effect.
 */
export const useScaleFade = () => {
  // Define the initial and final states for the scale and fade animation.
  // The animation will start with a scale of 0.5 (half the original size) and opacity of 0 (completely transparent)
  // and end with a scale of 1 (original size) and opacity of 1 (completely opaque).
  const scaleFadeAnimationState = useAnimationState({
    from: {
      scale: 0.9,
      opacity: 0,
    },
    to: {
      scale: 1,
      opacity: 1,
    },
  });

  // Return the scale and fade animation state.
  return scaleFadeAnimationState;
};

/**
 * Custom hook for creating a slide animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 to a translateX of 0.
 * This creates a slide-in effect.
 *
 * @returns An animation state that creates a slide-in effect.
 */
export const useSlide = () => {
  // Define the initial and final states for the slide animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and end with a translateX of 0 (original position).
  const slideAnimationState = useAnimationState({
    from: {
      translateY: -100,
    },
    to: {
      translateY: 0,
    },
  });

  // Return the slide animation state.
  return slideAnimationState;
};

/**
 * Custom hook for creating a slide and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 and opacity of 0 to a translateX of 0 and opacity of 1.
 * This creates a combined effect of sliding and fading in.
 *
 * @returns An animation state that creates a slide and fade-in effect.
 */
export const useSlideFade = () => {
  // Define the initial and final states for the slide and fade animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and opacity of 0 (completely transparent)
  // and end with a translateX of 0 (original position) and opacity of 1 (completely opaque).
  const slideFadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
      translateY: 10,
    },
    to: {
      opacity: 1,
      translateY: 0,
    },
  });

  // Return the slide and fade animation state.
  return slideFadeAnimationState;
};

/**
 * Custom hook for creating a collapse animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scaleY of 1 to a scaleY of 0.
 * This creates a collapse effect.
 *
 * @returns An animation state that creates a collapse effect.
 */
export const useCollapse = () => {
  // Define the initial and final states for the collapse animation.
  // The animation will start with a scaleY of 1 (original size) and end with a scaleY of 0 (collapsed).
  const collapseAnimationState = useAnimationState({
    from: {
      opacity: 0,
      height: 0,
    },
    to: {
      opacity: 1,
      height: 100,
    },
  });

  // Return the collapse animation state.
  return collapseAnimationState;
};
