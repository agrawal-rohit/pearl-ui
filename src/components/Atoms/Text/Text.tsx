import React from "react";
import { Text as RNText } from "react-native";
import responsiveSize from "../../../utils/responsiveSize";
import { useAtomicComponentConfig } from "../../../hooks/useAtomicComponentConfig";
import {
  backgroundColor,
  BackgroundColorProps,
  color,
  ColorProps,
  layout,
  LayoutProps,
  opacity,
  OpacityProps,
  spacing,
  SpacingProps,
  textShadow,
  TextShadowProps,
  typography,
  TypographyProps,
  visible,
  VisibleProps,
} from "../../../theme/src/styleFunctions";
import {
  ComponentSizes,
  ComponentVariants,
  ResponsiveValue,
  StyleFunctionContainer,
} from "../../../theme/src/types";
import { useTheme } from "../../../hooks/useTheme";

type TextStyleProps = ColorProps &
  BackgroundColorProps &
  OpacityProps &
  VisibleProps &
  LayoutProps &
  TypographyProps &
  SpacingProps &
  TextShadowProps & {
    /** The size of the text */
    size?: ResponsiveValue<ComponentSizes<"Text">>;
    /** The variant of the text */
    variant?: ResponsiveValue<ComponentVariants<"Text">>;
    /** Whether to slightly scale the font size based on the screen dimensions */
    scaleFontSize?: boolean;
  };

export const textStyleFunctions = [
  color,
  backgroundColor,
  opacity,
  visible,
  layout,
  typography,
  spacing,
  textShadow,
] as StyleFunctionContainer[];

type RNTextProps = React.ComponentProps<typeof RNText>;

export type TextProps = TextStyleProps &
  Omit<RNTextProps, keyof TextStyleProps>;

export const buildFontConfig = (
  textStyle: any,
  allowFontScaling: boolean
): object => {
  const fontWeight = textStyle.fontWeight;
  const fontStyle = textStyle.fontStyle || "normal";

  let fontSize = textStyle.fontSize;
  let lineHeight = textStyle.lineHeight;

  if (allowFontScaling) {
    fontSize = responsiveSize(fontSize);
    lineHeight = responsiveSize(lineHeight);
  }

  const initialFontFamily = textStyle.fontFamily;
  const { theme } = useTheme();

  const finalFontFamily =
    theme.fontConfig[initialFontFamily][fontWeight][fontStyle];

  return {
    fontSize,
    fontWeight,
    fontFamily: finalFontFamily,
    fontStyle,
  };
};

/**
 * Text is the component which controls the typography across your app. By default, it renders a <Text /> component
 */
const Text = React.forwardRef((rest: TextProps, ref: any) => {
  const props = useAtomicComponentConfig(
    "Text",
    rest,
    {
      size: rest.size,
      variant: rest.variant,
    },
    "primary",
    textStyleFunctions
  );

  const memoizedBuildFontConfig = React.useCallback(
    () => buildFontConfig(props.style, props.scaleFontSize),
    [props.style, props.scaleFontSize]
  );

  props.style = {
    includeFontPadding: false,
    ...props.style,
    ...memoizedBuildFontConfig(),
  };

  return (
    <RNText
      ref={ref}
      accessible={true}
      accessibilityRole="text"
      allowFontScaling
      {...props}
    >
      {props.children}
    </RNText>
  );
});

export default Text;
