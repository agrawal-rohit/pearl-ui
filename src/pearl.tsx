import {
  AtomComponent,
  ComponentTypes,
  FinalPearlTheme,
  MoleculeComponent,
  PearlComponent,
  StyleFunctionContainer,
} from "./theme/src/types";
import React from "react";
import { useAtomicComponentConfig } from "./hooks/useAtomicComponentConfig";
import { boxStyleFunctions, BoxStyleProps } from "./theme/src/style-functions";
import { useStyleProps } from "./hooks/useStyleProps";
import { useMolecularComponentConfig } from "./hooks/useMolecularComponentConfig";
import { MotiView, motify } from "moti";
import { useMotiWithStyleProps } from "./hooks/useMotiWithStyleProps";
import { useTheme } from "./hooks/useTheme";
import { Platform } from "react-native";

/**
 * Configuration object that adds metadata to a component that's required by Pearl UI
 */
interface pearlConfig {
  /**
   * The name of the component in the FinalPearlTheme
   */
  componentName: keyof FinalPearlTheme["components"];
  /**
   * The type of the component
   */
  type: ComponentTypes;
  /**
   * Whether the component is animatable or not
   */
  animatable?: boolean;
}

/**
 * Hook to convert a third-party component to a Pearl component that can be configured using style props, an atomic component configuration or a molecular component configuration.
 * @param Component The base component on which the pearlification has to be applied
 * @param config A configuration object that adds metadata to a component that's required by Pearl UI
 * @param styleFunctions The set of style functions that would define the style props that should be supported by the pearlified component
 * @param moleculeConfigOptions The options for configuring the molecular component
 * @returns
 */
export function pearl<
  ComponentProps,
  ComponentType extends ComponentTypes = "basic",
  ComponentAtoms extends Record<string, any> = Record<string, any>,
  StyleProps = BoxStyleProps,
  Animateable extends boolean = true,
>(
  Component: any,
  config: pearlConfig = {
    componentName: "None",
    type: "basic",
    animatable: true,
  },
  styleFunctions: StyleFunctionContainer[] = boxStyleFunctions,
  moleculeConfigOptions: {
    partForOverridenStyleProps?: string | undefined;
    partForOverridenNativeProps?: string | undefined;
    partForOverridenAnimationProps?: string | undefined;
  } = {}
) {
  let FinalComponent: any | undefined;
  const isAnimatable = !!config.animatable && config.type !== "molecule";

  if (isAnimatable) {
    const ConvertedClassComponent = class extends React.Component<any, any> {
      static displayName = `Pearl${
        Component.name ?? Component.displayName ?? `NoName`
      }`;

      render() {
        const { children, ...props } = this.props;
        return (
          <Component ref={(props as any).forwardedRef} {...props}>
            {children}
          </Component>
        );
      }
    };

    if (Platform.OS === "web" && config.componentName === "Box")
      FinalComponent = MotiView;
    else FinalComponent = motify(ConvertedClassComponent)();
  } else {
    FinalComponent = Component;
  }

  return React.memo(
    React.forwardRef(
      (
        {
          children,
          ...rest
        }: PearlComponent<ComponentProps, StyleProps, Animateable> &
          AtomComponent<(typeof config)["componentName"], ComponentType> &
          MoleculeComponent<(typeof config)["componentName"], ComponentType> & {
            atoms?: Partial<ComponentAtoms>;
            children?: string | JSX.Element | JSX.Element[] | React.ReactNode;
          },
        ref: any
      ) => {
        const { colorMode } = useTheme();

        let convertedProps;
        if (config.type === "atom") {
          convertedProps = useAtomicComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions
          );
          if (config.animatable)
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
        } else if (config.type === "molecule") {
          convertedProps = useMolecularComponentConfig(
            config["componentName"],
            rest,
            {
              size: (rest as any).size,
              variant: (rest as any).variant,
            },
            rest.colorScheme,
            styleFunctions,
            moleculeConfigOptions.partForOverridenStyleProps,
            moleculeConfigOptions.partForOverridenNativeProps,
            moleculeConfigOptions.partForOverridenAnimationProps
          );
        } else {
          convertedProps = useStyleProps(rest, styleFunctions);
          if (config.animatable) {
            convertedProps = useMotiWithStyleProps(
              convertedProps,
              styleFunctions
            );
          }
        }

        return (
          <FinalComponent
            {...convertedProps}
            key={
              Platform.OS === "web" &&
              ["Text", "Icon", "Box"].includes(config.componentName as string)
                ? (convertedProps as any).key!
                  ? `${(convertedProps as any).key!}-${colorMode}`
                  : `${colorMode}`
                : undefined
            }
            ref={ref}
          >
            {children}
          </FinalComponent>
        );
      }
    )
  );
}
