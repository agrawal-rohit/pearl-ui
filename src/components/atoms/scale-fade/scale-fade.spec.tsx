import React from "react";
import { render } from "@testing-library/react-native";
import ScaleFade from "./scale-fade";
import Text from "../text/text";
import { ThemeProvider } from "../../../theme/src/theme-context";

jest.useFakeTimers();

describe("Atoms/ScaleFade", () => {
  it("passes the snapshot test when show is true", () => {
    const tree = render(
      <ThemeProvider>
        <ScaleFade show={true}>
          <Text>This is a test</Text>
        </ScaleFade>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test when show is false", () => {
    const tree = render(
      <ThemeProvider>
        <ScaleFade show={false}>
          <Text>This is a test</Text>
        </ScaleFade>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test when initialScale is set", () => {
    const tree = render(
      <ThemeProvider>
        <ScaleFade show={true} initialScale={0.5}>
          <Text>This is a test</Text>
        </ScaleFade>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
