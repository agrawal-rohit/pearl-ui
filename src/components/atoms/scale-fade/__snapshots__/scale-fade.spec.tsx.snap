// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/ScaleFade passes the snapshot test when initialScale is set 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "opacity": 1,
        "scale": 1,
      }
    }
    animatedStyle={
      {
        "value": {
          "opacity": 0,
          "transform": [
            {
              "scale": 0.5,
            },
          ],
        },
      }
    }
    collapsable={false}
    exit={
      {
        "opacity": 0,
        "scale": 0.5,
      }
    }
    exitTransition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
    from={
      {
        "opacity": 0,
        "scale": 0.5,
      }
    }
    style={
      {
        "opacity": 0,
        "transform": [
          {
            "scale": 0.5,
          },
        ],
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      animate={
        {
          "color": "#1A2138",
        }
      }
      animatedStyle={
        {
          "value": {
            "color": "#1A2138",
          },
        }
      }
      collapsable={false}
      style={
        {
          "color": "#1A2138",
          "fontFamily": "Poppins-Regular",
          "fontSize": 16.099999999999998,
          "fontStyle": "normal",
          "fontWeight": "400",
          "includeFontPadding": false,
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      This is a test
    </Text>
  </View>
</View>
`;

exports[`Atoms/ScaleFade passes the snapshot test when show is false 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
/>
`;

exports[`Atoms/ScaleFade passes the snapshot test when show is true 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "opacity": 1,
        "scale": 1,
      }
    }
    animatedStyle={
      {
        "value": {
          "opacity": 0,
          "transform": [
            {
              "scale": 0.7,
            },
          ],
        },
      }
    }
    collapsable={false}
    exit={
      {
        "opacity": 0,
        "scale": 0.7,
      }
    }
    exitTransition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
    from={
      {
        "opacity": 0,
        "scale": 0.7,
      }
    }
    style={
      {
        "opacity": 0,
        "transform": [
          {
            "scale": 0.7,
          },
        ],
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <Text
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={true}
      animate={
        {
          "color": "#1A2138",
        }
      }
      animatedStyle={
        {
          "value": {
            "color": "#1A2138",
          },
        }
      }
      collapsable={false}
      style={
        {
          "color": "#1A2138",
          "fontFamily": "Poppins-Regular",
          "fontSize": 16.099999999999998,
          "fontStyle": "normal",
          "fontWeight": "400",
          "includeFontPadding": false,
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      This is a test
    </Text>
  </View>
</View>
`;
